### PR TITLE
New listFile convenience methods

### PR DESCRIPTION
**Pull request by agur**
- Added more listFile convenience methods.
- There are additional usability improvements possible, just added a few for now.

### CODE REVIEW

**Comment by amirgur1** [comment]
- Created a JIRA ticket [IO-473](https://issues.apache.org/jira/browse/IO-473) for this one per these [instructions](https://github.com/apache/commons-io/blob/trunk/CONTRIBUTING.md).
Yet my commit and comment did not yet include a `IO-473:` prefix.
- Yet not seeing too many pull requests.  Thus will also email this to the dev list.
- Also I can't assign [IO-473](https://issues.apache.org/jira/browse/IO-473) on my name for fixing.  Maybe needs to be a contributor.

**Comment by kinow** [comment]
Hey @amirgur1 ! Thanks for your contribution. I have edited the issue in JIRA, but since I am not really familiar with [io] I will let others review and comment on your pull request. Thanks a lot!

File: src/main/java/org/apache/commons/io/FileUtils.java [hunk 1]
```diff
@@ -486,6 +486,42 @@ private static void innerListFiles(final Collection<File> files, final File dire
         }
     }
 
+    /** @see #listFiles(String[])  */
```
Comment (britter) [inline src/main/java/org/apache/commons/io/FileUtils.java]: Every public method should have JavaDocs explaining the behavior.

**Comment by britter** [comment]
Please add some JUnit tests for the new functionality

**Comment by krosenvold** [comment]
This pr will require unit tests to be merged !

**Comment by garydgregory** [comment]
-1 needs unit tests. See also discussion items in the Jira ticket.

**Comment by agur** [comment]
> -1 needs unit tests. See also discussion items in the Jira ticket.

Mentioned on Jira, adding tests isn't hard and agree it's needed.  But also see Jira discussion and can further cont. there.  To ensure we do add this if it's indeed a missing functionality.

**Pull request closed without merging**